src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndTreatment/index.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import Title from 'Assets/Elements/Title';
+import { StyledFormGroup } from 'Assets/Elements/StyledFormGroup';
+import { StyledSelectTemplateButton } from 'Assets/Elements/StyledSelectTempleteButton';
+import { useTranslation } from 'react-i18next';
+import CustomizedTextField from 'Assets/Elements/CustomizedTextField';
+import { Grid } from '@material-ui/core';
+import { StyledDivider } from '../Style';
+import { useFormContext } from 'react-hook-form';
+const DiagnosisAndTreatment = () => {
+  const { t } = useTranslation();
+  const {
+    permission,
+    setPopUpProps,
+    setValue,
+    register,
+    watch,
+  } = useFormContext();
+
+  const diagnosisAndTreatmentFields = watch([
+    'findingsDetails',
+    'diagnosisDetails',
+    'treatmentDetails',
+  ]);
+  const callBack = (data, name) => {
+    setValue(name, data);
+  };
+
+  const handlePopUpProps = (title, fields, id, callBack, name) => {
+    setPopUpProps((prevState) => {
+      return {
+        ...prevState,
+        popupOpen: true,
+        formFieldsTitle: title,
+        formFields: fields,
+        formID: id,
+        setTemplatesTextReturned: callBack,
+        name,
+      };
+    });
+  };
+  return (
+    <StyledFormGroup>
+      <Title
+        label={t('Diagnosis and treatment')}
+        fontSize='22px'
+        bold
+        color='#000b40'
+      />
+      <StyledDivider />
+      <Grid
+        container
+        direction='row'
+        justify='flex-start'
+        alignItems='baseline'>
+        <CustomizedTextField
+          name='findingsDetails'
+          inputRef={register}
+          label={t('Findings details')}
```

Comment:
Sorry, it was not clear, I want we move the field name to constant for all the inputs with StyledSelectTemplateButton.

Here also